tests_phi/testcases/login_functions.py

```diff
@@ -0,0 +1,65 @@
+from tests_phi.testcases.test_base import TestBase
+from tests_phi.utils.common_methods import CommonMethods
+from tests_phi.pages.login_page import LoginPage
+from tests_phi.pages.dashboard_page import DashboardPage
+from tests_phi.utils import constants
+from selenpy.support import browser
+import logging
+import pytest
+
+class Test_LoginTests(TestBase):
+    
+    login_page = LoginPage()
+    dashboard_page = DashboardPage()
+    common_methods = CommonMethods()
+    incorrect_login_message = "Username or password is invalid"
+    
+    @pytest.mark.usefixtures("clean_up_tc001")
+    def test_da_login_tc001(self):
+        '''Verify that user can login specific repository successfully via Dashboard login page with correct credentials'''
+        
+        logging.info("1. Navigate to login page")
+        self.login_page.go_to_login_page()
+         
+        logging.info("2. Login with valid account")
+        self.login_page.login(constants.USERNAME,"")
+        self.login_page.wait_for_login_page_is_not_visible()
+         
+        logging.info("VP: Dashboard page displays")
+        assert "Execution Dashboard" in browser.get_driver().title
+         
+    def test_da_login_tc002(self):
+        '''Verify that user fails to login specific repository successfully via Dashboard login page with incorrect credentials'''
+        
+        incorrect_password = "abc"
+        
+        logging.info("1. Navigate to login page")
+        self.login_page.go_to_login_page()
+        
+        logging.info("2. Login with invalid account")
+        self.login_page.login(constants.USERNAME+"Incorrect", incorrect_password)
+        
+        logging.info("VP: '%s' message displays"% self.incorrect_login_message)
+        assert browser.get_alert().text == self.incorrect_login_message
+        browser.close_alert()
+ 
+    def test_da_login_tc003(self):
+        '''Verify that user fails to log in specific repository successfully via Dashboard login page with correct username and incorrect password'''
+        
+        incorrect_password = "abc"
```

Comment:
`incorrect_password `should be defined as constant.

Reply:
updated